django/contrib/auth/decorators.py

```diff
@@ -50,6 +45,32 @@ def login_required(function=None, redirect_field_name=REDIRECT_FIELD_NAME, login
     return actual_decorator
 
 
+def superuser_required(redirect_field_name=REDIRECT_FIELD_NAME, login_url=None):
+    """
+    Decorator for views that checks that the user is superuser, redirecting
+    to the log-in page if necessary or disallowing non superusers.
+    """
+
+    def decorator(view_func):
+        @wraps(view_func)
+        def _wrapped_view(request, *args, **kwargs):
+            if request.user.is_superuser:
+                return view_func(request, *args, **kwargs)
+
+            if not request.user.is_authenticated:
+                return _get_login_redirect(
+                    request,
+                    redirect_field_name=redirect_field_name,
+                    login_url=login_url
+                )
+
+            raise PermissionDenied
```

Comment:
Raising `PermissionDenied` could potentially leak private data. For some sites, the existence of a URL itself should not be public knowledge. I think it would be better to make this a 404 so -- to the user -- it looks like the URL doesn't exist.